tests/system/test_query.py

```diff
@@ -29,15 +444,35 @@ def test_dry_run(bigquery_client: bigquery.Client, scalars_table: str):
     assert len(query_job.schema) > 0
 
 
-def test_session(bigquery_client: bigquery.Client):
+def test_query_error_w_api_method_query(bigquery_client: bigquery.Client):
+    """No job is returned from jobs.query if the query fails."""
+
+    with pytest.raises(exceptions.NotFound, match="not_a_real_dataset"):
+        bigquery_client.query(
+            "SELECT * FROM not_a_real_dataset.doesnt_exist", api_method="QUERY"
+        )
+
+
+def test_query_error_w_api_method_insert(bigquery_client: bigquery.Client):
+    """With jobs.insert, an exception is thrown when fetching the results.."""
+
+    query_job = bigquery_client.query("SELECT * FROM not_a_real_dataset.doesnt_exist")
```

Comment:
Maybe rename the test case to explain that we also test that "INSERT" is indeed the default?

Reply:
Renamed and added more explanation to the comment that this (error only on `result()`) is the "old" behavior and is maintained for backwards compatibility.